Blog post: Camel Quarkus 1.0.0

ppalaga opened the pull request:
An early version. Any feedback is welcome!

ppalaga commented on content/blog/2020/08/camel-quarkus-release-1.0.0/index.md:
```diff
@@ -0,0 +1,77 @@
+---
+title: "Camel Quarkus 1.0.0 Released"
+date: 2020-07-07
```
Yes, thanks for catching it! A copy-and-paste problem :)

zregvart replied:
I usually take one from [unsplash](https://unsplash.com/s/photos/camel)

ppalaga commented:
> a featured image is missing

Any hints where to get one?

zregvart commented:
> > a featured image is missing
> 
> Any hints where to get one?

Have a look at [unsplash](https://unsplash.com/s/photos/camel)

jamesnetherton commented on content/blog/2020/08/camel-quarkus-release-1.0.0/index.md:
```diff
@@ -0,0 +1,77 @@
+preview: What's new in Camel Quarkus 1.0.0
+---
+
+Apache Camel community celebrates the release 1.0.0 of Camel Quarkus!
```
```suggestion
The Apache Camel community celebrates the 1.0.0 release of Camel Quarkus!
```

ppalaga replied:
+1

jamesnetherton commented on content/blog/2020/08/camel-quarkus-release-1.0.0/index.md:
```diff
@@ -0,0 +1,77 @@
+* Easy to start with: either via [code.quarkus.io](https://code.quarkus.io/) or using `mvn quarkus:create`
+* The "live coding" feature, a.k.a. `mvn compile quarkus:dev` shortens your development iterations
+* Easy to test using JUnit 5 extensions
+* Shields from the complexity of the native compilation using [GraalVM](https://www.graalvm.org/) or
```
```suggestion
* Hides the complexity of the native compilation using [GraalVM](https://www.graalvm.org/) or
```

ppalaga replied:
Thanks!

jamesnetherton commented on content/blog/2020/08/camel-quarkus-release-1.0.0/index.md:
```diff
@@ -0,0 +1,77 @@
+
+TODO:
+bootstrapping methods
+reflectionless
```
> reflectionless

We can't really make that claim yet as https://github.com/apache/camel-quarkus/issues/1108 wont be fixed until we get an updated Camel release.

ppalaga replied:
Indeed, let me remove it.

ppalaga commented:
5f484b7

* Removed the history paragraph
* Fixed the issues found by @oscerd  and @jamesnetherton 
* More text
* Good to merge from my PoV

Please re-review

zregvart commented:
We need #455 merged for the build to succeed.

davsclaus commented on content/blog/2020/08/camel-quarkus-release-1.0.0/index.md:
```diff
@@ -0,0 +1,131 @@
+* Hides the complexity of the native compilation using [GraalVM](https://www.graalvm.org/) or
+  [Mandrell](https://github.com/graalvm/mandrel)
+
+Check our https://camel.apache.org/camel-quarkus/latest/user-guide/first-steps.html[User guide]!
```
In preview then this link looked broken.

davsclaus commented on content/blog/2020/08/camel-quarkus-release-1.0.0/index.md:
```diff
@@ -0,0 +1,131 @@
+## What is Camel Quarkus?
+
+Camel Quarkus brings the outstanding integration capabilities of Apache Camel to [Quarkus](https://quarkus.io/) - the
+toolkit for writing subatomically small and supersonically fast Java, Kotlin and Scala applications. In addition to
```
I am not so sure Quarkus is well known for fast Scala applications. I dont think the Scala community think of Quarkus as a Scala framework they are using. I dont think you can find Scala mentioned on the quarkus website.

I would remove Scala.

ppalaga replied:
OK, let's remove Scala. Although there is scala extension and scala should work, it is not especially promoted from the Quarkus side.

davsclaus commented on content/blog/2020/08/camel-quarkus-release-1.0.0/index.md:
```diff
@@ -0,0 +1,131 @@
+Both methods configure and start a Camel Context for you. The main difference is in how much the resulting Camel Context
+obeys the conventions known from Camel on other platforms. If you like to configure your routes using
+`camel.component.*` properties, then you need to add the `camel-quarkus-main` dependency and the properties
+will work as usual on Camel standalone or Camel SpringBoot. Of course, this brings some overhead. You may prefer this
```
Spring Boot (there is a space between)

davsclaus replied:
Of course this brings some overhead? What overhead and why of course. From end user perspectice this may actually be easier. Its more flexible and works better for externalized configuration.

I would tone down this as it seems you are saying "dont do this".

ppalaga replied:
The additional dependency brings both build time and runtime overhead. That's true for almost any dependency and that's why I said 'of course'. I admit I do not have any exact data and I admit the overhead might be negligible for basic applications. IIRC, less overhead was one of the motivations for not including main by default. I fully second the UX argument, and based on that, I was rather opposing not including main by default. It is definitely good that we discuss how to present this option. Let me think how to formulate it better.

lburgazzoli replied:
I think this just boils down how you want to use camel-quarkus: as a library or as an application runtime.

IMHO, camel-quarkus-core is for embedding camel functionalities with full control, whereas camel-quarkus-main is mainly for application.

ppalaga replied:
I decided to remove the sentence. Thanks for the feedback.

aldettinger commented:
@ppalaga well done :+1 Looks good overall. I proposed changes in the 2nd commit, feel free to take what you like.

davsclaus commented on content/blog/2020/08/camel-quarkus-release-1.0.0/index.md:
```diff
@@ -0,0 +1,131 @@
+will work as usual on Camel standalone or Camel SpringBoot. Of course, this brings some overhead. You may prefer this
+option if you used Camel before or if you are migrating from other platform.
+
+If you rather come from the Quarkus side and you never wrote Camel integrations before, you may choose to use the bare
```
I would say from CDI side - its not really Quarkus side. CDI is a bit "coding heavy" in terms of configuration as this requires Java coding and adding annotations and whatnot.

It reads as if it is two "big" alternatives, that you cannot use CDI with Camel Main, and vice versa.

ppalaga replied:
> I would say from CDI side

Good idea.

> It reads as if it is two "big" alternatives, that you cannot use CDI with Camel Main, and vice versa.

Good point, the text should not awake that impression.

ppalaga replied:
> > It reads as if it is two "big" alternatives, that you cannot use CDI with Camel Main, and vice versa.
> 
> Good point, the text should not awake that impression.

Adding `You can configure Camel using CDI even if you have Camel Main in your application.`

davsclaus commented on content/blog/2020/08/camel-quarkus-release-1.0.0/index.md:
```diff
@@ -0,0 +1,131 @@
+
+However, the native compilation with GraalVM is all but easy and straightforward. First, there are
+[limitations](https://github.com/oracle/graal/blob/master/substratevm/LIMITATIONS.md) for what your code and all your
+dependencies may or may not do. Second, the native compiler needs to be configured through many options that may
```
may or may not be easy -> not easy

I dont think anything with graalvm is easy for the average user.

ppalaga replied:
Yeah, shorter is better.

zregvart commented on content/blog/2020/08/camel-quarkus-release-1.0.0/index.md:
```diff
@@ -0,0 +1,130 @@
+preview: What's new in Camel Quarkus 1.0.0
+---
+
+<sub><sup>Original image by <a href="https://commons.wikimedia.org/wiki/User:99of9">Toby Hudson</a> <a href="https://creativecommons.org/licenses/by-sa/3.0">CC BY-SA 3.0</a> via <a href="https://en.wikipedia.org/wiki/Camel_racing#/media/File:CamelRacingCamelCup2009Heat.JPG">Wikipedia</a></sup></sub>
```
Might make sense to add [manual summary](https://gohugo.io/content-management/summaries/#front-matter-summary), this bit ends up in the [blog post list](https://deploy-preview-454--camel.netlify.app/blog/).

ppalaga replied:
Let me add a summary

ppalaga commented:
54e8265: addressed the comments of @davsclaus and adopted all changes (thanks!) from @aldettinger

ppalaga commented:
b6f138e: added a summary

ppalaga commented on content/blog/2020/08/camel-quarkus-release-1.0.0/index.md:
```diff
@@ -0,0 +1,132 @@
+authors: ["ppalaga"]
+categories: ["Releases", "Camel Quarkus"]
+preview: The first stable release of Camel Quarkus
+summary: Walk through the highlights of the first stable release: Developer joy, Camel component coverage, Bootstrap, CDI, native mode and more!
```
@zregvart I hope this would work as a summary?

zregvart replied:
Looks [good](https://deploy-preview-454--camel.netlify.app/blog/).

ppalaga commented:
1c79c88: added double quotes to summary value
